src/Containers.jsx

```diff
@@ -617,64 +663,71 @@
-
-            if (this.props.ownerFilter !== "all") {
-                filtered = filtered.filter(id => {
-                    if (this.props.ownerFilter === "user")
-                        return this.props.containers[id].uid === null;
-                    return this.props.containers[id].uid === this.props.ownerFilter;
-                });
-            }
-
-            if (this.props.textFilter.length > 0) {
-                const lcf = this.props.textFilter.toLowerCase();
-                filtered = filtered.filter(id => this.filterContainersByText(lcf, id));
-            }
-
-            // Remove infra and service containers
-            filtered = filtered.filter(id => !this.props.containers[id].IsInfra && !this.props.containers[id].IsService);
-
-            const getHealth = id => {
-                const state = this.props.containers[id]?.State;
-                return state?.Health?.Status || state?.Healthcheck?.Status;
-            };
+        if (isLoaded) {
+            const filtered = this.filterContainers(this.props.containers);
+            pods = { ...this.props.pods };
+            const podServiceNameIdMap = {};
+
+            const running_pod_quadlets = new Set();
+            Object.values(this.props.pods).forEach(pod => {
+                const service_name = pod?.Labels?.PODMAN_SYSTEMD_UNIT;
+                // FIXME: what about --user and system with the same unit name, make it a dict?
+                if (service_name) {
+                    running_pod_quadlets.add(utils.makeKey(pod.uid, service_name));
+                    podServiceNameIdMap[service_name] = utils.makeKey(pod.uid, pod.Id);
+                }
+            });
 
-            filtered.sort((a, b) => {
-                // Show unhealthy containers first
-                const a_health = getHealth(a);
-                const b_health = getHealth(b);
-                if (a_health !== b_health) {
-                    if (a_health === "unhealthy")
-                        return -1;
-                    if (b_health === "unhealthy")
-                        return 1;
+            Object.keys(this.props.quadletPods).forEach(key => {
+                const values = this.props.quadletPods[key];
+                if (!running_pod_quadlets.has(utils.makeKey(values.uid, values.Labels.PODMAN_SYSTEMD_UNIT))) {
+                    pods[key] = values;
                 }
-                // User containers are in front of system ones
-                if (this.props.containers[a].uid !== this.props.containers[b].uid)
-                    return (this.props.containers[a].uid === 0) ? 1 : -1;
-                return this.props.containers[a].Name > this.props.containers[b].Name ? 1 : -1;
             });
 
-            Object.keys(this.props.pods || {}).forEach(pod => { partitionedContainers[pod] = [] });
+            Object.keys(pods || {}).forEach(pod => { partitionedContainers[pod] = [] });
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2266-e8a12e1c-20250917-134350-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)